### ISSUE TITLE
Pyright not recognizing metaclass keyword arguments

### ISSUE THREAD

**Issue by StockerMC** [microsoft/pyright#2436]
**Describe the bug**
Pyright doesn't recognize the keyword arguments of a metaclass when passing them in the class definition, though it does for classes with `__init_subclass__` defined.

**Screenshots or Code**
```python
from typing import TypeVar, Tuple, Dict, Any, Type, TYPE_CHECKING

CMT = TypeVar('CMT', bound='CommandMeta')

class BaseCommand:
    if TYPE_CHECKING:
        __foo__: int
        __bar__: str

    def __init_subclass__(cls, *, foo: int, bar: str) -> None: ...

class CommandMeta(type):
    if TYPE_CHECKING:
        __foo__: int
        __bar__: str

    def __new__(
        cls: Type[CMT],
        cls_name: str,
        bases: Tuple[type, ...],
        attrs: Dict[str, Any],
        *,
        foo: int,
        bar: str,
    ) -> CMT: ...


class One(BaseCommand, foo=1): ...  # Argument missing for parameter "bar"
class Two(metaclass=CommandMeta, foo=''): ...  # No error for the incorrect type or the missing parameter
```

**VS Code extension or command-line**
Are you running pyright as a VS Code extension or a command-line tool?
Command line (version 1.1.178)

**Comment by erictraut** [comment]
Looks like this is a known missing feature in mypy. [This issue](https://github.com/python/mypy/issues/2653) was filed almost five years ago, and it still hasn't been addressed.

I think that metaclasses that take typed keyword parameters are pretty rare, so it's understandable that such a feature would be low priority for a type checker.

For now I'm going to close this request, but we can will reconsider in the future if there is sufficient interest (additional requests or upvotes on this issue).

**Comment by Zomatree** [comment]
I think this should be added, python uses them in the stdlib<sup>[1]</sup> and i commonly use them in my code as they can remove boilerplate.

I dont think we should be pinning not adding features due to mypy not supporting them, mypy is notorious for being slow at adding new features and fixing bugs.

[1]: https://docs.python.org/3/library/typing.html#typing.TypedDict

**Comment by erictraut** [comment]
This is included in pyright 1.1.201, which I just published. It will also be included in the next release of pylance.
